tests/pipeline/test_pipeline.py

```diff
@@ -118,12 +120,92 @@ def test_substitute_component_names(self, snapshot: SnapshotTest):
         enriched_pipeline = yaml.safe_load(result.stdout)
 
         assert isinstance(enriched_pipeline, dict)
+
+        # calculate pipeline name with the exact same code that kpops uses
+        path_without_file = (
+            Path(RESOURCE_PATH / "component-type-substitution/pipeline.yaml")
+            .resolve()
+            .relative_to(Path(PIPELINE_BASE_DIR).resolve())
+            .parts[:-1]
+        )
+        if not path_without_file:
+            raise ValueError("The pipeline-base-dir should not equal the pipeline-path")
+        pipeline_name = "-".join(path_without_file)
+
+        assert (
+            enriched_pipeline["components"][0]["name"]
+            == pipeline_name + "-scheduled-producer"
```

Comment:
can't we assert this using a static string or does it always change?

Reply:
It shouldn't change, will fix it

Reply:
done